client/components/Layout/Layout.tsx

```diff
@@ -28,10 +30,44 @@ type Props = LayoutOptions & {
 };
 
 const Layout = (props: Props) => {
-	const { locationData, loginData, communityData } = usePageContext();
+	const { locationData, loginData, communityData, pageData } = usePageContext();
 	const { blocks, isNarrow, layoutPubsByBlock, id = '', collection } = props;
 	const pubsByBlockId = resolveLayoutPubsByBlock(layoutPubsByBlock, blocks);
 
+	const { page } = useAnalytics();
+
+	/** Page track should only be done on mount & on client side */
+	useEffectOnce(() => {
+		/** Only track actual page visits, not dashboard visits e.g. when editing a page */
+		if (locationData.isDashboard) {
+			return;
+		}
+
+		const payload = collection
+			? {
+					type: 'collection' as const,
+					communityId: collection.communityId,
+					title: collection.title,
+					collectionId: collection.id,
+					collectionTitle: collection.title,
+					collectionSlug: collection.slug,
+			  }
+			: pageData
+			  ? {
+						type: 'page' as const,
+						communityId: communityData.id,
+						pageSlug: pageData.slug,
+						title: pageData.title,
+			    }
+			  : undefined;
```

Comment:
Might be worth a comment describing when this `undefined` case would occur. And better yet if there's a way to isolate that case to an `if` statement above it would simplify this ternary a little.